plugins/modules/s3_object.py

```diff
@@ -591,49 +604,73 @@ def delete_key(module, s3, bucket, obj):
         s3.delete_object(Bucket=bucket, Key=obj)
         module.exit_json(msg="Object deleted from bucket %s." % (bucket), changed=True)
     except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed while trying to delete %s." % obj)
-
+        raise S3ObjectFailure("Failed while trying to delete %s." % obj, e)
 
-def create_dirkey(module, s3, bucket, obj, encrypt, expiry):
-    if module.check_mode:
-        module.exit_json(msg="PUT operation skipped - running in check mode", changed=True)
+def put_object_acl(module, s3, bucket, obj, params={}):
     try:
-        params = {'Bucket': bucket, 'Key': obj, 'Body': b''}
-        if encrypt:
-            params['ServerSideEncryption'] = module.params['encryption_mode']
-        if module.params['encryption_kms_key_id'] and module.params['encryption_mode'] == 'aws:kms':
```

Comment:
You cannot actually default params on `{}` like this. Well, actually you actually can, but you risk some annoying side-effects.

See:
https://docs.python-guide.org/writing/gotchas/